huaweicloud/services/modelarts/resource_huaweicloud_modelarts_resource_pool.go

```diff
@@ -393,20 +752,39 @@ func resourceModelartsResourcePoolUpdate(ctx context.Context, d *schema.Resource
 
 func buildUpdateResourcePoolBodyParams(d *schema.ResourceData) map[string]interface{} {
 	bodyParams := map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"annotations": map[string]interface{}{
-				"os.modelarts/description": utils.ValueIngoreEmpty(d.Get("description")),
-			},
-		},
-		"spec": map[string]interface{}{
-			"scope":     d.Get("scope"),
-			"resources": buildResourcePoolResourceFlavor(d.Get("resources")),
-		},
+		"metadata": buildUpdateResourcePoolMetaDataBodyParams(d),
+		"spec":     buildUpdateResourcePoolSpecBodyParams(d),
 	}
 	return bodyParams
 }
 
-func buildResourcePoolResourceFlavor(rawParams interface{}) []map[string]interface{} {
+func buildUpdateResourcePoolMetaDataBodyParams(d *schema.ResourceData) map[string]interface{} {
+	params := map[string]interface{}{
+		"annotations": buildUpdateResourcePoolMetaDataAnnotationsBodyParams(d),
+	}
+	return params
+}
+
+func buildUpdateResourcePoolMetaDataAnnotationsBodyParams(d *schema.ResourceData) map[string]interface{} {
+	params := map[string]interface{}{
+		"os.modelarts/description": utils.ValueIngoreEmpty(d.Get("description")),
+	}
+	if d.Get("charging_mode") == "prePaid" {
+		params["os.modelarts/order.id"] = ""
```

Comment:
can we delete this line?

Reply:
no, an empty value is required when update the resource